test/OperationTest.cpp

```diff
@@ -323,37 +340,35 @@ TEST(Operation, verifyRuntimeInformationIsUpdatedForLazyOperations) {
   auto result = valuesForTesting.runComputation(
       timer, ComputationMode::LAZY_IF_SUPPORTED);
 
-  EXPECT_EQ(valuesForTesting.runtimeInfo().status_,
-            RuntimeInformation::Status::lazilyMaterialized);
-  EXPECT_EQ(valuesForTesting.runtimeInfo().totalTime_, 0ms);
-  EXPECT_EQ(valuesForTesting.runtimeInfo().originalTotalTime_, 0ms);
-  EXPECT_EQ(valuesForTesting.runtimeInfo().originalOperationTime_, 0ms);
+  auto& rti = valuesForTesting.runtimeInfo();
+
+  EXPECT_EQ(rti.status_, RuntimeInformation::Status::lazilyMaterialized);
+  EXPECT_EQ(rti.totalTime_, 0ms);
+  EXPECT_EQ(rti.originalTotalTime_, 0ms);
+  EXPECT_EQ(rti.originalOperationTime_, 0ms);
 
   auto& idTables = result.idTables();
 
   auto iterator = idTables.begin();
   ASSERT_NE(iterator, idTables.end());
 
-  EXPECT_EQ(valuesForTesting.runtimeInfo().status_,
-            RuntimeInformation::Status::lazilyMaterialized);
-  EXPECT_EQ(valuesForTesting.runtimeInfo().numCols_, 2);
-  EXPECT_EQ(valuesForTesting.runtimeInfo().numRows_, 1);
+  EXPECT_EQ(rti.status_, RuntimeInformation::Status::lazilyMaterialized);
+  EXPECT_EQ(rti.numCols_, 2);
+  EXPECT_EQ(rti.numRows_, 1);
 
   ++iterator;
   ASSERT_NE(iterator, idTables.end());
 
-  EXPECT_EQ(valuesForTesting.runtimeInfo().status_,
-            RuntimeInformation::Status::lazilyMaterialized);
-  EXPECT_EQ(valuesForTesting.runtimeInfo().numCols_, 2);
-  EXPECT_EQ(valuesForTesting.runtimeInfo().numRows_, 2);
+  EXPECT_EQ(rti.status_, RuntimeInformation::Status::lazilyMaterialized);
+  EXPECT_EQ(rti.numCols_, 2);
+  EXPECT_EQ(rti.numRows_, 2);
 
   ++iterator;
   ASSERT_EQ(iterator, idTables.end());
 
-  EXPECT_EQ(valuesForTesting.runtimeInfo().status_,
-            RuntimeInformation::Status::lazilyMaterialized);
-  EXPECT_EQ(valuesForTesting.runtimeInfo().numCols_, 2);
-  EXPECT_EQ(valuesForTesting.runtimeInfo().numRows_, 2);
+  EXPECT_EQ(rti.status_, RuntimeInformation::Status::lazilyMaterialized);
+  EXPECT_EQ(rti.numCols_, 2);
+  EXPECT_EQ(rti.numRows_, 2);
```

Comment:
Can't you use your `expectAtEachStageOfGenerator` here
 + a small helper lambda that checks status + numCols + numRows
 (can be used 4 times here).

Reply:
Only 3 times so the benefit is not that big actually :)